app/controllers/events/view/tickets/orders.js

```diff
@@ -0,0 +1,46 @@
+import Controller from '@ember/controller';
+import { run } from '@ember/runloop';
+
+export default Controller.extend({
+  isLoadingcsv: false,
+
+  isLoadingpdf: false,
+
+  actions: {
+    export(mode) {
+      this.set(`isLoading${mode}`, true);
+      this.get('loader')
+        .load(`/events/${this.get('model.id')}/export/orders/${mode}`)
+        .then(exportJobInfo => {
+          this.requestLoop(exportJobInfo);
+        })
+        .catch(() => {
+          this.set(`isLoading${mode}`, false);
+          this.get('notify').error(this.get('l10n').t('An unexpected error has occurred.'));
+        });
+    }
+  },
+  requestLoop(exportJobInfo) {
+    run.later(() => {
+      this.get('loader')
+        .load(exportJobInfo.task_url, { withoutPrefix: true })
+        .then(exportJobStatus => {
+          if (exportJobStatus.state === 'SUCCESS') {
+            window.location = exportJobStatus.result.download_url;
+            this.get('notify').success(this.get('l10n').t('Download Ready'));
+          } else if (exportJobStatus.state === 'WAITING') {
+            this.requestLoop(exportJobInfo);
+            this.get('notify').alert(this.get('l10n').t('Task is going on.'));
+          } else {
+            this.get('notify').error(this.get('l10n').t('CSV Export has failed.'));
```

Comment:
@mayank8318 Just noticed that, this action will be common for both `csv` and `pdf` exports, so don't hardcode it.